agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/common/NetworkFriendlyExceptions.java

```diff
@@ -67,19 +103,45 @@ public static boolean logSpecialOneTimeFriendlyException(
 
   private static String getSslFriendlyMessage(String url) {
     return FriendlyException.populateFriendlyMessage(
-        getSslFriendlyExceptionBanner(url),
-        getSslFriendlyExceptionAction(url),
         "Unable to find valid certification path to requested target.",
+        getSslFriendlyExceptionAction(url),
+        getFriendlyExceptionBanner(url),
+        "This message is only logged the first time it occurs after startup.");
+  }
+
+  private static String getCipherFriendlyMessage(String url, List<String> missingCiphers) {
+    return FriendlyException.populateFriendlyMessage(
+        "Probable root cause may be : missing cipher suites which are expected by the requested target.",
+        getCipherFriendlyExceptionAction(url, missingCiphers),
+        getFriendlyExceptionBanner(url),
         "This message is only logged the first time it occurs after startup.");
   }
 
-  private static String getSslFriendlyExceptionBanner(String url) {
-    if (url.equals(DefaultEndpoints.LIVE_ENDPOINT)) {
+  private static String getFriendlyExceptionBanner(String url) {
+    if (url.contains(DefaultEndpoints.LIVE_ENDPOINT)) {
       return "ApplicationInsights Java Agent failed to connect to Live metric end point.";
     }
     return "ApplicationInsights Java Agent failed to send telemetry data.";
   }
 
+  private static String getCipherFriendlyExceptionAction(String url, List<String> missingCiphers) {
+    StringBuilder actionBuilder = new StringBuilder();
+    actionBuilder.append("The following are the cipher suites from Java runtime: ").append("\n");
+    if (missingCiphers.size() > 0) {
+      for (String missingCipher : missingCiphers) {
+        actionBuilder.append(missingCipher).append("\n");
+      }
+    }
+    actionBuilder
+        .append(
+            "Please add the required java modules to include the missing cipher suites that are expected from the target endpoint:"
```

Comment:
this is not the only reason why cipher suite may be missing, I think better to just have generic message here